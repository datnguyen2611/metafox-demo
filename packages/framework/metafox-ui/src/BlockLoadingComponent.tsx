/**
 * @type: service
 * name: BlockLoading
 */
import { LoadingComponentProps } from '@metafox/framework';
import { getErrString } from '@metafox/utils';
import { Alert, CircularProgress, Box } from '@mui/material';
import { styled } from '@mui/material/styles';
import React from 'react';

const BlockLoadingBackdrop = styled(Box, {
  name: 'BlockLoading',
  slot: 'Backdrop'
})({
  position: 'absolute',
  left: 0,
  top: 0,
  bottom: 0,
  right: 0
});

const BlockLoadingContent = styled(Box, {
  name: 'BlockLoading',
  slot: 'Content'
})({
  position: 'absolute',
  left: '50%',
  top: '50%',
  marginLeft: -20,
  marginTop: -20
});

export default function BlockLoadingComponent({
  error,
  size = 40,
  sx
}: LoadingComponentProps) {
  const msg = getErrString(error);

  return (
    <BlockLoadingBackdrop sx={sx}>
      <BlockLoadingContent>
        {error ? (
          <Alert variant="filled" color="error" children={msg} />
        ) : (
          <CircularProgress color="primary" size={size} />
        )}
      </BlockLoadingContent>
    </BlockLoadingBackdrop>
  );
}
