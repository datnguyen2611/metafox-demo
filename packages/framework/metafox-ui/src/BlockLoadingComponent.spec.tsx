import * as React from 'react';
import renderer from 'react-test-renderer';
import BlockLoadingComponent from './BlockLoadingComponent';

describe('BlockLoadingComponent', () => {
  it('render no err', async () => {
    const loading = renderer.create(<BlockLoadingComponent />);
    expect(loading.toJSON()).toMatchSnapshot();
  });
  it('render err', async () => {
    const loading = renderer.create(
      <BlockLoadingComponent error="error server" />
    );
    expect(loading.toJSON()).toMatchSnapshot();
  });
});
