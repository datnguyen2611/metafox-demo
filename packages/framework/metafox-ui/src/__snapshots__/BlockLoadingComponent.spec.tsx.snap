// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockLoadingComponent render err 1`] = `
<div
  className="css-6zdhku-BlockLoading-backdrop"
>
  <div
    className="css-pwygrr-BlockLoading-content"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-filledError MuiAlert-filled css-1c3trtb-MuiPaper-root-MuiAlert-root"
      role="alert"
    >
      <div
        className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
          data-testid="SuccessOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
          />
        </svg>
      </div>
      <div
        className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
      >
        e
      </div>
    </div>
  </div>
</div>
`;

exports[`BlockLoadingComponent render no err 1`] = `
<div
  className="css-6zdhku-BlockLoading-backdrop"
>
  <div
    className="css-pwygrr-BlockLoading-content"
  >
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={{}}
        />
      </svg>
    </span>
  </div>
</div>
`;
